source/agora/consensus/EnrollmentManager.d

```diff
@@ -423,6 +423,35 @@ public class EnrollmentManager
         return false;
     }
 
+    /***************************************************************************
+
+        Get the unregistered enrollments in the block
+
+        Params:
+            enrolls = will contain the unregistered enrollments data if found
+
+        Returns:
+            Return true if the unregistered enrollment data was found
+
+    ***************************************************************************/
+
+    public bool getEnrollments (out Enrollment[] enrolls) @trusted
+    {
+        enrolls.length = 0;
+        auto results = this.db.execute("SELECT val FROM validator_set" ~
+            " WHERE enrolled_height is null");
+
+        foreach (row; results)
+            enrolls ~= deserializeFull!Enrollment(row.peek!(ubyte[])(0));
+
+        if (enrolls.length > 0)
+        {
+            enrolls.sort!("a.utxo_key < b.utxo_key");
```

Comment:
I think we could move this to the query, couldn't we ?

Reply:
I thought I could solve it with Query.
In DB, the key is saved as Byte[]
Then it should encode again and order it into Query.
Would it be better to do this?

Reply:
So the issue is that SQLite is not able to ORDER BY on binary data ?

Reply:
If it's the same sort order of binary data, it's easy to fix with order by.

Reply:
Binary data sorted by ORDER BY key ASC by SQLite
````
[[8, 100, 43, 116, 108, 238, 60, 244, 157, 180, 124, 36, 129, 44, 190, 193, 5, 116, 71, 54, 226, 28, 8, 142, 30, 105, 99, 149, 121, 32, 87, 150, 10, 225, 220, 62, 61, 232, 246, 135, 128, 228, 1, 206, 244, 145, 169, 118, 18, 35, 160, 227, 250, 81, 105, 202, 67, 138, 97, 190, 6, 47, 212, 172, 233, 218, 93, 19, 142, 198, 86, 11, 46, 114, 36, 170, 94, 10, 25, 83, 246, 149, 255, 50, 93, 146, 75, 197, 221, 11, 57, 19, 118, 138, 16, 144, 100, 93, 85, 134, 179, 56, 57, 23, 86, 178, 171, 24, 120, 125, 63, 129, 123, 181, 3, 56, 230, 252, 21, 98, 198, 148, 59, 101, 131, 223, 246, 70, 253, 240, 3, 175, 175, 12, 12, 228, 70, 10, 224, 252, 223, 82, 189, 201, 110, 162, 206, 250, 22, 128, 241, 31, 176, 132, 125, 193, 187, 252, 102, 13, 32, 224, 85, 113, 141, 200, 146, 115, 79, 15, 42, 92, 30, 216, 199, 40, 56, 254, 146, 43, 129, 123, 123, 244, 130, 73, 154, 105, 178, 114, 24, 234, 60, 25, 7]]
[[91, 149, 84, 94, 175, 174, 191, 52, 18, 102, 193, 30, 46, 120, 106, 107, 106, 51, 211, 38, 108, 207, 170, 165, 116, 42, 78, 1, 254, 29, 141, 56, 150, 194, 10, 8, 109, 124, 3, 161, 230, 138, 131, 38, 56, 31, 14, 212, 47, 7, 70, 62, 182, 142, 54, 35, 250, 239, 86, 213, 76, 62, 1, 137, 214, 136, 116, 211, 119, 81, 183, 202, 106, 135, 81, 28, 87, 202, 252, 248, 84, 135, 97, 161, 16, 47, 180, 77, 219, 215, 210, 177, 144, 162, 223, 143, 80, 37, 106, 62, 210, 35, 19, 252, 64, 193, 95, 139, 218, 17, 45, 65, 160, 163, 234, 53, 175, 224, 89, 23, 180, 212, 168, 215, 180, 140, 213, 62, 253, 240, 3, 98, 233, 7, 164, 12, 106, 239, 62, 26, 54, 248, 136, 238, 212, 187, 90, 97, 160, 24, 224, 54, 168, 236, 249, 82, 92, 198, 99, 2, 225, 157, 216, 152, 201, 247, 180, 43, 239, 71, 183, 225, 221, 111, 253, 84, 100, 178, 216, 207, 205, 198, 18, 94, 48, 85, 229, 102, 83, 27, 69, 67, 92, 10, 9]]
[[125, 122, 52, 234, 186, 48, 238, 210, 214, 233, 49, 156, 97, 59, 50, 128, 152, 166, 17, 93, 133, 253, 189, 243, 94, 177, 24, 58, 116, 176, 215, 84, 114, 169, 98, 119, 209, 137, 144, 42, 151, 58, 25, 133, 153, 79, 249, 86, 8, 246, 133, 205, 20, 45, 208, 149, 218, 227, 158, 178, 6, 133, 38, 77, 143, 253, 111, 148, 226, 4, 21, 164, 69, 86, 46, 74, 98, 83, 248, 49, 174, 35, 35, 164, 218, 194, 122, 221, 99, 106, 11, 248, 87, 186, 168, 104, 48, 228, 244, 44, 211, 56, 233, 181, 183, 173, 191, 47, 2, 68, 9, 92, 248, 146, 99, 96, 2, 122, 245, 243, 143, 3, 128, 113, 190, 137, 137, 49, 253, 240, 3, 61, 59, 153, 129, 139, 202, 184, 39, 59, 149, 92, 64, 205, 191, 199, 209, 12, 229, 25, 13, 174, 208, 154, 21, 198, 171, 195, 70, 187, 111, 22, 241, 25, 116, 76, 6, 26, 22, 220, 78, 78, 192, 1, 42, 244, 203, 85, 4, 23, 224, 222, 72, 51, 1, 8, 98, 233, 11, 124, 168, 116, 121, 46, 7]]
```
deserialize from Binary data 
````
0xacd42f06be618a43ca6951fae3a0231276a991f4ce01e48087f6e83d3edce10a965720799563691e8e081ce236477405c1be2c81247cb49df43cee6c742b6408
0x89013e4cd556effa23368eb63e46072fd40e1f3826838ae6a1037c6d080ac296388d1dfe014e2a74a5aacf6c26d3336a6b6a782e1ec1661234bfaeaf5e54955b
0x4d268506b29ee3da95d02d14cd85f60856f94f9985193a972a9089d17762a97254d7b0743a18b15ef3bdfd855d11a69880323b619c31e9d6d2ee30baea347a7d
````
Not sorted in order of desired UTXOkey hash.

Reply:
Oh, I see. The binary data for SQLlite was little endian.

Reply:
I made a new function to reverse bytes before putting it in to DB.
It is also used in SELECT.
@Geod24 Please check to see if it's okay.